packages/sdk-js/src/handlers/auth.ts

```diff
@@ -94,4 +129,26 @@ export class AuthHandler {
 			await this.axios.post('/auth/password/reset', { token, password });
 		},
 	};
+
+	/**
+	 * There is no prettier way to do this. We need to set access token before first request.
+	 * This way we intercept axios request and only first time request token from store,
+	 * and allows us to do new Directus(url).items(col).read() without having to handle
+	 * access_token restoration in methods
+	 */
+	private async initializeAccessToken(config: AxiosRequestConfig) {
+		if (!this.accessTokenInitialized) {
+			this.accessTokenInitialized = true;
+			const token = await this.storage.getItem('directus_access_token');
+			console.log(11);
+
+			if (token) {
+				console.log(21);
+				this.token = token;
+				config.headers.Authorization = `Bearer ${token}`;
```

Comment:
This isn't needed. The setter for token initializes the header in axios

Reply:
It's not the same. Changing this.token in `this.axios.interceptors.request.use` does not changes token for current request.
So if user starts app with `new SDK(...).items('posts').read()` setting `this.token` would not set access token for existing request

Here is the code with console.log added
![Screenshot from 2020-12-16 22-40-34](https://user-images.githubusercontent.com/19491605/102409728-da789000-3fef-11eb-9b50-af4f533ee634.png)
Here is console
![Screenshot from 2020-12-16 22-41-42](https://user-images.githubusercontent.com/19491605/102409825-fc721280-3fef-11eb-9e00-68da75dc7ac4.png)

Reply:
Ah right, that ties back into my other comment then 🙂  https://github.com/directus/directus/pull/3337#discussion_r544592396

Reply:
Yeah, added `eject` in `initializeAccessToken`